From 6733df3e242a39668c7853ca86119061a8b98b6d Mon Sep 17 00:00:00 2001
From: vladsendrix <83285656+vladsendrix@users.noreply.github.com>
Date: Fri, 31 Dec 2021 03:18:04 +0100
Subject: [PATCH 16/17] PixelPropsUtils: Spoof Pixel XL for Google Photos

pixel_2016_exclusive.xml is already added, why not have it actually work

Change-Id: If0b8064c7cb633139ee7c30a7bf1a8543bcc9cad
---
 .../internal/util/custom/PixelPropsUtils.java    | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/core/java/com/android/internal/util/custom/PixelPropsUtils.java b/core/java/com/android/internal/util/custom/PixelPropsUtils.java
index 4f5ea8d17938..d0f928c70131 100644
--- a/core/java/com/android/internal/util/custom/PixelPropsUtils.java
+++ b/core/java/com/android/internal/util/custom/PixelPropsUtils.java
@@ -40,6 +40,11 @@ public class PixelPropsUtils {
             "com.google.android.apps.recorder"
     };
 
+    private static final Map<String, Object> propsToChangePixelXL;
+    private static final String[] packagesToChangePixelXL = {
+            "com.google.android.apps.photos"
+    };
+
     private static final Map<String, ArrayList<String>> propsToKeep;
     private static final String[] extraPackagesToChange = {
             "com.android.chrome",
@@ -69,6 +74,13 @@ public class PixelPropsUtils {
         propsToChangePixel5.put("PRODUCT", "redfin");
         propsToChangePixel5.put("MODEL", "Pixel 5");
         propsToChangePixel5.put("FINGERPRINT", "google/redfin/redfin:12/SQ1A.211205.008/7888514:user/release-keys");
+        propsToChangePixelXL = new HashMap<>();
+        propsToChangePixelXL.put("BRAND", "google");
+        propsToChangePixelXL.put("MANUFACTURER", "Google");
+        propsToChangePixelXL.put("DEVICE", "marlin");
+        propsToChangePixelXL.put("PRODUCT", "marlin");
+        propsToChangePixelXL.put("MODEL", "Pixel XL");
+        propsToChangePixelXL.put("FINGERPRINT", "google/marlin/marlin:10/QP1A.191005.007.A3/5972272:user/release-keys");
     }
 
     public static void setProps(String packageName) {
@@ -85,6 +97,10 @@ public class PixelPropsUtils {
                 propsToChange = propsToChangePixel5;
             }
 
+            if (Arrays.asList(packagesToChangePixelXL).contains(packageName)) {
+                propsToChange = propsToChangePixelXL;
+            }
+
             if (DEBUG) Log.d(TAG, "Defining props for: " + packageName);
             for (Map.Entry<String, Object> prop : propsToChange.entrySet()) {
                 String key = prop.getKey();
-- 
2.32.0

